src/Application.php

```diff
@@ -639,33 +468,15 @@ private function checkForDuplicateRoute($path, $methods = null)
      */
     private function emitMarshalServerRequestException($exception)
     {
-        $response = (new Response())
-            ->withStatus(StatusCode::STATUS_BAD_REQUEST);
-        $finalHandler = $this->getFinalHandler();
-        $response = $finalHandler(new ServerRequest(), $response, $exception);
+        if ($this->container && $this->container->has(Middleware\ErrorResponseGenerator::class)) {
```

Comment:
I understand where you're coming from, but I still disagree.

`Zend\Stratigility\Middleware\ErrorHandler` is _middleware_, and its public methods are only around:

- attaching listeners
- handling requests

To make it behave as you suggest would change the purpose of the `ErrorHandler` class: it is intended to be _middleware_, allowing developers to replace the service with _other middleware_. If we make the API change you suggest, we'd effectively be indicating that the class must implement _some other interface_, one that defines the `handleStartupException()` method. It's supposed to be _just middleware_.

My point in re-using the `ErrorResponseGenerator` is that it has a defined interface: a callable accepting the exception, a request, and a response, and returns a response. It's an instance called *by* the `ErrorHandler`, and, because it's a service, can be re-used elsewhere, which is what I've done here.

What that means is that you could provide your own ErrorResponseGenerator implementation entirely if you feel the need to handle startup exceptions due to inability to marshal a request in the same way as other application exceptions. In such a case, I'd likely attach my listeners to *that* instance, instead of the `ErrorHandler`, as the `ErrorResponseGenerator` could then notify listeners instead.

I personally feel that this is an edge case; there are very few situations where this should happen, and the only ones I can think of off the top of my head are when using async systems, which have to marshal the request information manually instead of relying on the SAPI. As such, the liklihood that you'd ever need to log such an error is pretty slim.